src/components/product/product.service.ts

```diff
@@ -381,11 +380,14 @@ export class ProductService {
           id: produces.p.properties.id,
           createdAt: produces.p.properties.createdAt,
           __typename: (ProducibleType as any)[typeName],
-          ...(await this.getProducibleByType(
-            produces.p.properties.id,
-            typeName,
-            session
-          )),
+          // scriptureReferencesOverride is not null or an empty array.
+          scriptureReferences: !scriptureReferencesOverride.length
```

Comment:
We want an empty array here to take precedence over the producible's scripture. null and empty array have to be handled differently - otherwise we wouldn't even both with the nulls 😄